src/test/java/mate/academy/service/UserServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package mate.academy.service;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.dao.UserDao;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.impl.UserServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class UserServiceTest {
+    private static final String EMAIL = "someemail@gmail.com";
+    private static final String PASSWORD = "password";
+    private static final Role.RoleName ROLE = Role.RoleName.USER;
+    private static UserDao userDao;
+    private static UserService userService;
+    private static User user;
+    private static Role role;
+
+    @BeforeAll
+    static void beforeAll() {
+        userDao = Mockito.mock(UserDao.class);
+        PasswordEncoder passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        userService = new UserServiceImpl(userDao, passwordEncoder);
+
+        role = new Role();
+        user = new User();
+    }
+
+    @BeforeEach
+    void setUp() {
+        role.setRoleName(ROLE);
+
+        user.setEmail(EMAIL);
+        user.setPassword(PASSWORD);
+        user.setRoles(Set.of(role));
+    }
+
+    @Test
+    void save_Ok() {
+        Mockito.when(userDao.save(user)).thenReturn(user);
+
+        User actual = userService.save(user);
+        Assertions.assertNotNull(actual, "Method must return User object");
+        Assertions.assertEquals(user, actual,
+                "Expected " + user + ", but was " + actual);
+    }
```

Comment:
Such test cases are not useful in production. You are just mocking the single and unique line in the UserService and checking the mocked result.